### PR TITLE
[Dy2Stat]Support non-tensor type in `input_spec`

### PR DESCRIPTION
**Pull request by Aurelius84**
### PR types

New features
### PR changes

APIs
### Describe

[Dy2Stat]Support non-tensor type in `input_spec`

### CODE REVIEW

**Review by TCChenlong — DISMISSED** [review]
LGTM

**Comment by CtfGo** [comment]
LGTM

File: python/paddle/fluid/dygraph/dygraph_to_static/utils.py [hunk 1]
```diff
@@ -1421,10 +1422,10 @@ def input_specs_compatible(src_input_specs, desired_input_specs):
     Returns True if the two input specs are compatible, otherwise False.
 
     args:
-        src_input_spec (list[InputSpec]|tuple(InputSpec)): list/tuple of
-            paddle.static.InputSpec
-        desired_input_specs (list[InputSpec]|tuple(InputSpec)): list/tuple of
-            paddle.static.InputSpec
+        src_input_spec (list or tuple[InputSpec et.al]): list/tuple of
+            paddle.static.InputSpec or or int/str et.al
```
Comment (chenwhql) [inline python/paddle/fluid/dygraph/dygraph_to_static/utils.py]: two `or`?

File: python/paddle/fluid/dygraph/dygraph_to_static/utils.py [hunk 2]
```diff
@@ -1421,10 +1422,10 @@ def input_specs_compatible(src_input_specs, desired_input_specs):
...
+        desired_input_specs (list or tuple[InputSpec et.al]): list/tuple of
+            paddle.static.InputSpec or or int/str et.al
```
Comment (chenwhql) [inline python/paddle/fluid/dygraph/dygraph_to_static/utils.py]: same as above

**Review by chenwhql — COMMENTED** [review]
LGTM

**Review by hbwx24 — DISMISSED** [review]
LGTM

**Review by hbwx24 — COMMENTED** [review]
LGTM

File: python/paddle/fluid/dygraph/dygraph_to_static/utils.py (hunk 1 above)
Comment (Aurelius84) [reply python/paddle/fluid/dygraph/dygraph_to_static/utils.py]: thanks, fixed

File: python/paddle/fluid/dygraph/dygraph_to_static/utils.py (hunk 2 above)
Comment (Aurelius84) [reply python/paddle/fluid/dygraph/dygraph_to_static/utils.py]: thanks, fixed

**Review by TCChenlong — APPROVED** [review]

**Review by chenwhql — APPROVED** [review]

**Pull request merged**